In [1]:
from transformers import AutoModel, AutoTokenizer
import torch

model = AutoModel.from_pretrained("google/gemma-1.1-2b-it", use_auth_token="<>")
tokenizer = AutoTokenizer.from_pretrained("google/gemma-1.1-2b-it", use_auth_token="<>")

# Text to be tokenized
text = "My name is Ajay!"

# Encode text
input_ids = tokenizer.encode(text, add_special_tokens=True)

# Output the token IDs
print("Token IDs:", input_ids)

# Convert token IDs back to raw tokens and output them
raw_tokens = [tokenizer.decode([token_id]) for token_id in input_ids]
print("Raw tokens:", raw_tokens)

# Convert list of IDs to a tensor
input_ids_tensor = torch.tensor([input_ids])

# Pass the input through the model
with torch.no_grad():
    outputs = model(input_ids_tensor)

# Extract the embeddings
embeddings = outputs.last_hidden_state

# Print the embeddings
print("Embeddings: ", embeddings)

/usr/local/lib/python3.10/dist-packages/transformers/models/auto/auto_factory.py:466: FutureWarning: The `use_auth_token` argument is deprecated and will be removed in v5 of Transformers. Please use `token` instead.
  warnings.warn(


Loading checkpoint shards:   0%|          | 0/2 [00:00<?, ?it/s]

/usr/local/lib/python3.10/dist-packages/transformers/models/auto/tokenization_auto.py:720: FutureWarning: The `use_auth_token` argument is deprecated and will be removed in v5 of Transformers. Please use `token` instead.
  warnings.warn(


Token IDs: [2, 2926, 1503, 603, 164476, 235341]
Raw tokens: ['<bos>', 'My', ' name', ' is', ' Ajay', '!']
Embeddings:  tensor([[[ 0.2486, -0.7862, -0.2441,  ..., -0.6646, -0.4929, -0.6996],
         [ 0.6401,  1.4105,  0.4291,  ..., -0.2684,  0.0316, -1.2565],
         [-1.2397,  1.6598, -0.9212,  ...,  0.1642, -0.9899, -1.1751],
         [-0.8512,  0.2834, -2.4947,  ..., -0.1198,  1.3421, -1.1525],
         [-1.3338,  0.8203, -1.4401,  ...,  0.3993, -0.3663, -1.0987],
         [-1.3155,  0.2018, -0.4188,  ...,  0.2751, -0.3506, -0.5765]]])


Token IDs: [2, 2926, 1503, 603, 164476, 235341]

Raw tokens: ['<bos>', 'My', ' name', ' is', ' Ajay', '!']

Embeddings:  tensor([[[ 0.2486, -0.7862, -0.2441,  ..., -0.6646, -0.4929, -0.6996],
         [ 0.6401,  1.4105,  0.4291,  ..., -0.2684,  0.0316, -1.2565],
         [-1.2397,  1.6598, -0.9212,  ...,  0.1642, -0.9899, -1.1751],
         [-0.8512,  0.2834, -2.4947,  ..., -0.1198,  1.3421, -1.1525],
         [-1.3338,  0.8203, -1.4401,  ...,  0.3993, -0.3663, -1.0987],
         [-1.3155,  0.2018, -0.4188,  ...,  0.2751, -0.3506, -0.5765]]])